pkg/stanza/docs/operators/udp_input.md

```diff
@@ -50,11 +50,13 @@ for other encodings available.
 
 If set, the `async` configuration block instructs the `udp_input` operator to read and process logs asynchronsouly and concurrently.
 
-**note** If `async` is not set at all, a single thread will read lines synchronously.
+**note** If `async` is not set at all, a single thread will read & process lines synchronously.
 
 | Field                                   | Default              | Description |
 | ---                                     | ---                  | ---         |
-| `readers`                               | 1                    | Concurrency level - Determines how many go routines read from UDP port (and process logs before sending downstream). |
+| `readers`                               | 1                    | Concurrency level - Determines how many go routines read from UDP port and push to channel (to be handled by processors). |
+| `processors`                            | 1                    | Concurrency level - Determines how many go routines read from channel (pushed by readers) and process logs before sending downstream. |
+| `max_queue_length`                      | 100                  | Determines max length of channel being used by async reader routines. When channel reaches max number, reader routine will block until channel has room. |
```

Comment:
This could be a little clearer in regards to whether the allocation is per-reader or shared. Generally, I think it's best to avoid language specific terminology if possible, since many users may not know golang well. I think the following would be a bit clearer, if I've correctly myself.

```suggestion
| `max_queue_length`                      | 100                  | Determines max number of messages which may be waiting for a processor. While the queue is full, incoming messages will be dropped by readers. |
```

Reply:
changed description (your suggestion wasn't 100% accurate).